File: spiral_matrix.rs
pub fn spiral_matrix(size: u32) -> Vec<Vec<u32>> {
    let mut left: i32 = 0;
    let mut right: i32 = size as i32 - 1;
    let mut top: i32 = 0;
    let mut bottom: i32 = size as i32 - 1;
    let mut cnt = 1;
    let mut v = vec![vec![0; size as usize]; size as usize];
    while left <= right && top <= bottom {
        for i in left..=right {
            v[left as usize][i as usize] = cnt;
            cnt += 1;
        }
        top += 1;
        for i in top..=bottom {
            v[i as usize][right as usize] = cnt;
            cnt += 1;
        }
        right -= 1;
        for i in (left..=right).rev() {
            v[bottom as usize][i as usize] = cnt;
            cnt += 1;
        }
        bottom -= 1;
        for i in (top..=bottom).rev() {
            v[i as usize][left as usize] = cnt;
            cnt += 1;
        }
        left += 1;
    }
    v
}

File: solutions/python/spiral-matrix/1/spiral_matrix.py
def spiral_matrix(size):
    left, right = 0, size - 1
    top, bottom = 0, size - 1
    cnt = 1
    v = [[0] * size for _ in range(size)]
    while left <= right and top <= bottom:
        # Top row
        for i in range(left, right + 1):
            v[top][i] = cnt
            cnt += 1
        top += 1
        # Right column
        for i in range(top, bottom + 1):
            v[i][right] = cnt
            cnt += 1
        right -= 1
        # Bottom row
        if top <= bottom:
            for i in range(right, left - 1, -1):
                v[bottom][i] = cnt
                cnt += 1
            bottom -= 1
        # Left column
        if left <= right:
            for i in range(bottom, top - 1, -1):
                v[i][left] = cnt
                cnt += 1
            left += 1
    return v